pricelist_item_generator/models/pricelist.py

```diff
@@ -0,0 +1,426 @@
+    if not(len(vals) == 1 and 'todo' in vals):
+        self.price_generator_id.write({'to_update': True})
+        # some records should be flagged as 'update'
+        for object in self:
+            # if not yet created, no reason to update, we stay as 'create'
+            if object.todo != 'create':
+                todo_update |= object
+    return todo_update
+
+
+class PricelistItemGenerator(models.Model):
+    _name = "pricelist.item.generator"
+    _description = "Pricelist Items Generator"
+
+    sequence = fields.Integer(
+        string='Sequence', default=5,
+        help="Used to generate the sequence of the price item")
+    name = fields.Char(
+        string='Name', required=True,
+        help="Copied towards pricelist price item")
+    active = fields.Boolean(
+        string='Active', copy=False,
+        help="If checked, rules are exported towards "
+             "'Product pricelist items'")
+    to_update = fields.Boolean(
+        string='To update', readonly=True, default=True, copy=False,
+        help="Flag if the pricelist items needs to be build")
+    price_version_id = fields.Many2one(
+        comodel_name='product.pricelist.version',
+        string='Pricelist version', required=True,
+        domain="[('pricelist_id.type', '=', 'sale')]",
+        help="Only pricelist of 'Sale' type")
+    copy_product_condition = fields.Boolean(
+        string="Copy Involved Products", copy=False,
+        help="Check to copy 'involved products' in case of "
+             "duplication of generator")
+    copy_item_template = fields.Boolean(
+        string="Copy Price Items", copy=False,
+        help="Check to copy 'price items' in case of "
+             "duplication of generator")
+    product_condition_ids = fields.One2many(
+        comodel_name='pricelist.product.condition',
+        inverse_name='price_generator_id',
+        string='Product conditions',
+        help="In which products are applied the price elements")
+    item_template_ids = fields.One2many(
+        comodel_name='pricelist.item.template',
+        inverse_name='price_generator_id',
+        string='Pricelist item templates',
+        help="Pricelist item template")
+
+    _sql_constraints = [
+        ('name_uniq', 'unique(name)',
+         "'Name' field must be unique by generator"),
+    ]
+
+    @api.multi
+    def copy(self, default=None):
+        self.ensure_one()
+        default = dict(default or {})
+        default['name'] = '%s (Copy)' % self.name
+        # Conditional One2many copy
+        if self.copy_item_template:
+            default['item_template_ids'] = self._get_o2m_values(
+                self.item_template_ids, 'item_template_id')
+        if self.copy_product_condition:
+            default['product_condition_ids'] = self._get_o2m_values(
+                self.product_condition_ids, 'product_condition_id')
+        return super(PricelistItemGenerator, self).copy(default)
```

Comment:
It's a good practice to continue keeping keyword arguments as is, so please put `...copy(default=default)`